tests/net/lib/http_server/table/testcase.yaml

```diff
@@ -0,0 +1,8 @@
+tests:
+  # section.subsection
+  net.http.server.table:
```

Comment:
@rlubos - Emna & I weren't really sure what the tag / name should be here. I.e. "table" is just used everywhere for lack of a better term, but this is referring to the static HPACK table.

Reply:
I think a combination of `hpack` and `static_table` should be used. For the test, i"d use `hpack.static_table`. 

For the implementation, personally I'd:
 - rename `http_table.c` to `http_hpack.c`,
 - rename current APIs to use `static_table` instead of `table` (not to get confused with dynamic table which we don't implement yet)
 - `http_hpack.c` leaves a room for future extenstion with dynamic table, we should aslo conisder adding header parsing logic here  (according to https://datatracker.ietf.org/doc/html/rfc7541#section-6) as it's part of the HPACK spec.

Reply:
>  - rename current APIs to use `static_table` instead of `table` (not to get confused with dynamic table which we don't implement yet)

 Given that the static and dynamic table share the same key space, it's probably better _not_ to rename public API (i.e. functions) with `static_table`, since they would just need to be renamed again later if / when dynamic table support is added. It likely does make sense internally to differentiate between static and dynamic, of course.

https://httpwg.org/specs/rfc7541.html#index.address.space

What would also be good, @Emna-Rekik, is to create a static inline function in the header file to say whether a given key is static.

E.g.

```cpp
static inline bool http_hpack_key_is_static(int32_t key)
{
    return key >= 1 && key <= 61; /* use symbolic name */
}
```

The nice thing about eventually supporting a dynamic table (which would be trivial with a hash_map), is that there is only a simple constant offset (±62) required to convert from the common key space to the dynamic key space.